sycl/doc/developer/Debugging.md

```diff
@@ -0,0 +1,167 @@
+# Debugging DPC++
+
+## Building DPC++ in debug mode
+
+To build DPC++ in debug mode you can simply pass `-t Debug` to `configure.py`,
+however debug builds can be quite large and slow so using the following `CMake`
+options may help:
+
+- `-DLLVM_USE_SPLIT_DWARF=ON`: Use `-gsplit-dwarf`, this splits some of the
+  debug information out of the object files into their own separate files,
+  which reduces the size of the object files the linker has to load (see
+  [DebugFission](https://gcc.gnu.org/wiki/DebugFission)).
+- `-DLLVM_PARALLEL_LINK_JOBS=4`: Reduce the number of link jobs running in
+  parallel, to avoid running out of RAM when linking large debug build objects.
+- `-DLLVM_USE_LINKER=lld`: Use the LLVM linker `lld` instead of the GNU linker
+  `ld` as `lld` is usually faster.
+- `-DLIBCLC_CUSTOM_LLVM_TOOLS_BINARY_DIR=<path/to/release/build/bin>`: Use a
+  separate release build of DPC++ as the compiler for building the bitcode
+  libraries (`libclc` and `libdevice`), it normally uses the compiler from the
+  same build, but debug clang is quite slow so using a separate release clang
+  can be worth it.
```

Comment:
Another useful option worth mentioning:

LLVM_OPTIMIZED_TABLEGEN:BOOL
If enabled and building a debug or asserts build, the CMake build system will generate a Release build tree to build a fully optimized tablegen for use during the build. Enabling this option can significantly speed up build times, especially when building LLVM in Debug configurations.